File: webapp/php/FuxFramework/Http/FuxResponse.php
<?php

namespace Fux\Http;

class FuxResponse implements \JsonSerializable
{

    const ERROR = 'ERROR';
    const SUCCESS = 'OK';
    const CONFIRM = 'CONFIRM';

    protected $response = [];
    protected $canBePretty = false;

    public function __construct($status = null, $message = null, $data = null, $canBePretty = false)
    {
        if ($status !== null) $this->response['status'] = $status;
        if ($message !== null) $this->response['message'] = $message;
        if ($data !== null) $this->response['data'] = $data;
        $this->canBePretty = $canBePretty;
    }

    public function __toString()
    {
        return json_encode($this->response);
    }

    public function isOk()
    {
        return $this->response['status'] == self::SUCCESS;
    }

    public function isError()
    {
        return $this->response['status'] == self::ERROR;
    }

    public function isConfirm()
    {
        return $this->response['status'] == self::CONFIRM;
    }

    public function isPretty()
    {
        return $this->canBePretty;
    }

    public function getMessage()
    {
        return $this->response['message'] ?? null;
    }

    public function setMessage($message)
    {
        $this->response['message'] = $message;
    }

    public function getData()
    {
        return $this->response['data'] ?? null;
    }

    public function setData($data)
    {
        $this->response['data'] = $data;
    }

    public function getStatus()
    {
        return $this->response['status'] ?? null;
    }

    public function setStatus($status)
    {
        $this->response['status'] = $status;
    }

    public function jsonSerialize(): mixed
    {
        return $this->response;
    }

    /**
     * @param $array = [
     *      "status" => "ERROR" | "OK", //Optional
     *      "message" => "ab cd", //Optional
     *      "data" => mixed //Optional
     * ]
     * @return FuxResponse
     */
    public static function fromArray($array)
    {
        return new FuxResponse($array["status"] ?? null, $array["message"] ?? null, $array["data"] ?? null);
    }

    /**
     * @return FuxResponse
     */
    public static function success($message = null, $data = null)
    {
        return new FuxResponse(FuxResponse::SUCCESS, $message, $data);
    }

    /**
     * @return FuxResponse
     */
    public static function error($message = null, $data = null)
    {
        return new FuxResponse(FuxResponse::ERROR, $message, $data);
    }

}

File: cat/plugins/food_assistant/fux_response.py
import json


class FuxResponse:
    ERROR = 'ERROR'
    SUCCESS = 'OK'
    CONFIRM = 'CONFIRM'

    def __init__(self, status=None, message=None, data=None, can_be_pretty=False):
        self.response = {}
        if status is not None:
            self.response['status'] = status
        if message is not None:
            self.response['message'] = message
        if data is not None:
            self.response['data'] = data
        self.can_be_pretty = can_be_pretty

    def __str__(self):
        return json.dumps(self.response)

    def is_ok(self):
        return self.response.get('status') == self.SUCCESS

    def is_error(self):
        return self.response.get('status') == self.ERROR

    def is_confirm(self):
        return self.response.get('status') == self.CONFIRM

    def is_pretty(self):
        return self.can_be_pretty

    def get_message(self):
        return self.response.get('message')

    def set_message(self, message):
        self.response['message'] = message

    def get_data(self):
        return self.response.get('data')

    def set_data(self, data):
        self.response['data'] = data

    def get_status(self):
        return self.response.get('status')

    def set_status(self, status):
        self.response['status'] = status

    def json_serialize(self):
        return self.response

    @classmethod
    def from_dict(cls, dictionary):
        status = dictionary.get("status", None)
        message = dictionary.get("message", None)
        data = dictionary.get("data", None)
        return cls(status, message, data)

    @staticmethod
    def success(message=None, data=None):
        return FuxResponse(FuxResponse.SUCCESS, message, data)

    @staticmethod
    def error(message=None, data=None):
        return FuxResponse(FuxResponse.ERROR, message, data)
